Sources/SwiftFusion/Inference/BearingRangeFactor.swift

```diff
@@ -0,0 +1,153 @@
+// Copyright 2019 The SwiftFusion Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import TensorFlow
+
+/// Protocol for indicating a struct is range-capable
+public protocol RangeFunction: Differentiable {
+  associatedtype Base: Differentiable
+  associatedtype Target: Differentiable
+  @differentiable
+  static func range(_ from: Base, _ to: Target) -> Double
+}
+
+public struct PointToPointRange2D: RangeFunction {
```

Comment:
Note, to me it's weird that you talk about the "range" between two points. In GTSAM, the range is between a Pose2 (robot) and a Point2 (landmark). And, that should be in Pose2.

Reply:
This is to demonstrate RangeFunction, I will delete if you don't like this one

Reply:
Deleted